python/ray/external_storage.py

```diff
@@ -287,6 +287,9 @@ class ExternalStorageSmartOpenImpl(ExternalStorage):
     For example, if your uri is a local file path, you should pre-create
     the directory.
 
+    Currently, this plugin only supports S3. Please feel free to contribute
+    to the support for other storage systems.
```

Comment:
Why not just special case it for S3?

Reply:
Yeah actually that sounds like a better idea. Let me fix the comment